libyul/YulStack.cpp

```diff
@@ -349,12 +356,18 @@ YulStack::assembleEVMWithDeployed(std::optional<std::string_view> _deployName)
 			return {std::make_shared<evmasm::Assembly>(assembly), std::make_shared<evmasm::Assembly>(runtimeAssembly)};
 		}
 	}
+	catch (Error const& _error)
+	{
+		reportCodeGenerationError(_error);
+		return {nullptr, nullptr};
```

Comment:
If an error occurs, `assembleEVMWithDeployed` would return a tuple where `creationAssembly` is `nullptr`, causing the assertion on [line 265](https://github.com/ethereum/solidity/pull/15610/files#diff-56078b38b33a09bedce880b8e750bc4da72df11ae8600d9f1850d7aa658d7f01R265) to fail. Is this expected? Previously, this method always returned a shared pointer to Assembly, which now appears to conflict with the assertion.

```c++
auto [creationAssembly, deployedAssembly] = assembleEVMWithDeployed(_deployName);
yulAssert(creationAssembly, "");
```

Reply:
Good catch! I did not notice that and we don't currently have any exceptions that would happen in this part of the code so tests did not detect it either.

And I need to handle that `nullptr` in `CompilerStack` as well. This is actually a bit inconvenient, because I need to add some code for converting errors reported in Yul to Solidity ones and I'm not yet sure how to handle source locations (they point at Yul code).

Reply:
The underlying cause of this (`CompilerStack` ignoring errors from `YulStack`) is now fixed in #15624.